src/blocks/events/edit.js

```diff
@@ -0,0 +1,212 @@
+			jQuery( this.slickTarget ).slick( 'slickSetOption', 'infinite', false, true );
+		}
+	}
+
+	/**
+	 * Disconnect the MutationObserver and remove Slick carousels.
+	 */
+	componentWillUnmount() {
+		if ( !! this.props.attributes.externalCalendarUrl ) {
+			jQuery( this.slickTarget ).slick( 'unslick' );
+		}
+
+		this.observer.disconnect();
+	}
+
+	/**
+	 * The callback for our MutationObserver.
+	 *
+	 * A React Ref is used to track changes to our ServerSideRender component
+	 * in order to initialize Slick Carousel for the rendered content.
+	 *
+	 * @param {Array} mutationsList List of objects describing each change that occurred.
+	 */
+	mutationObserverCallback( mutationsList ) {
+		for ( const mutation of mutationsList ) {
+			if ( mutation.type === 'childList' && mutation.addedNodes.length > 0 ) {
+				if ( mutation.addedNodes[ 0 ].outerHTML.match( 'wp-block-coblocks-events' ) ) {
+					this.slickTarget = mutation.addedNodes[ 0 ].children[ 0 ];
+
+					jQuery( this.slickTarget ).slick( {
+						infinite: false,
+						rows: this.slickTarget.dataset.perPage,
+
+						// Slick settings to disable within the Block Editor to prevent conflicts.
+						accessibility: false, // Disable tabbing and arrow key navigation.
+						draggable: false, // Disable mouse dragging slides.
+						swipe: false, // Disable swiping slides.
+						touchMove: false, // Disable touch swiping slides.
+					} );
+				}
+			}
+		}
+	}
+
+	toggleExternalCalendarControls() {
+		const { showExternalCalendarControls } = this.state;
+
+		if ( ! showExternalCalendarControls === false ) {
+			this.props.setAttributes( { externalCalendarUrl: '' } );
+		}
+
+		this.setState( { showExternalCalendarControls: ! showExternalCalendarControls } );
+	}
+
+	saveExternalCalendarUrl() {
+		this.props.setAttributes( { externalCalendarUrl: this.state.externalCalendarUrl } );
+		this.setState( { isEditing: false } );
+	}
+
+	insertNewItem() {
+		const { clientId, innerBlocks } = this.props;
+		const newEvent = createBlock( 'coblocks/event-item' );
+		dispatch( 'core/block-editor' ).insertBlock( newEvent, innerBlocks.length, clientId );
+	}
+
+	render() {
+		const {
+			className,
+			attributes,
+			setAttributes,
+		} = this.props;
+
+		const toolbarControls = [
+			{
+				icon: 'edit',
+				title: __( 'Edit Calendar URL', 'coblocks' ),
+				onClick: () => this.setState( { isEditing: ! this.state.isEditing } ),
+			},
+		];
+
+		return (
+			<Fragment>
+				<InspectorControls
+					attributes={ attributes }
+					toggleExternalCalendarControls={ this.toggleExternalCalendarControls }
+					showExternalCalendarControls={ this.state.showExternalCalendarControls }
+					eventsRangeOptions={ EVENTS_RANGE_OPTIONS }
+					onChangeEventsToShow={ eventsToShow => setAttributes( { eventsToShow } ) }
+					onChangeEventsRange={ eventsRange => setAttributes( { eventsRange } ) }
+				/>
+
+				{ !! attributes.externalCalendarUrl &&
```

Comment:
We don't have to do it now, but in the future we should remove the toolbar and inspector controls into their own files.

Reply:
I only left it in because the code is so small right now.